src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -210,7 +210,7 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
                         walk(astWithComments, contents, AstState.WITH_COMMENTS);
                     }
                     final SortedSet<LocalizedMessage> filteredMessages =
```

Comment:
We do not need to loop over the set of localized messages in case the filters set is empty. Add the check. If filters set is empty, we should not invoke getFilteredMessages.